src/network/clientpackethandler.cpp

```diff
@@ -764,6 +764,39 @@ void Client::handleCommand_ItemDef(NetworkPacket* pkt)
 	m_itemdef_received = true;
 }
 
+
+void Client::handleCommand_Settings(NetworkPacket* pkt)
+{
+	infostream << "Client: Received settings: packet size: "
+			<< pkt->getSize() << std::endl;
+
+	u8 type;
+	u8 flags;
+	(*pkt) >> type;
+	assert(type == 0);
+	(*pkt) >> flags;
+	bool replace = (flags & 0x01) != 0;
+	if (replace)
+		m_settings_received = true;
+
+	// If managed by server, the settings have already been applied.
+	// No need to override (or actually: overriding is harmful)
+	if (g_settings_managed_by_server)
+		return;
+
+	// Decompress settings
+	std::string datastring(pkt->getString(2), pkt->getSize());
```

Comment:
no need to serialize/compress for a little amount of data.
send u16 count + array[count] {std::string key, std::string value }